tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -3919,6 +3919,10 @@ public synchronized String getName(
         if (localePattern == null) {
             localePattern = "{0} ({1})";
         }
+        // Hack
+        if (localeSeparator == null) {
```

Comment:
Hmmm. This should never happen. Can you see which locales this occurs in?

Reply:
It occurs in about three-quarters of locales, including major ones like `it`, `de`. See my comment about my suspicion that they aren't following fallback

Reply:
This is fixed per our discussion earlier!

We also switched to using `CalculatedCoverageLevels`, which cleans up a lot of code and output.

Does the PR look good now?